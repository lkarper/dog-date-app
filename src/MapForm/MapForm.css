.MapForm__coordinates-fieldset {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.MapForm__set-coordinates-button {
    margin: 20px auto;
}
