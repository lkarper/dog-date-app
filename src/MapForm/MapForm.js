import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Map from '../Map/Map';
import CurrentLocation from '../CurrentLocation/CurrentLocation';
import './MapForm.css';

const MapForm = (props) => {
    const { 
        coordinates, 
        setCoordinates, 
    } = props;

    const [tempCoordinates, setTempCoordinates] = useState(coordinates);

    return (
        <fieldset 
            className='MapForm__coordinates-fieldset sub-fieldset'
        >
            <legend>Set coordinates for your meeting here</legend>
            <p>Using this feature will make it easier for others to find the location you set, but it is not required.</p>
            <Map 
                tempCoordinates={tempCoordinates} 
                setTempCoordinates={setTempCoordinates} 
            />
            <button 
                className='MapForm__set-coordinates-button button'
                type='button' 
                onClick={() => setCoordinates(tempCoordinates)}
            >
                Set meeting place to above coordinates
            </button>
            <CurrentLocation 
                coordinates={coordinates}
            />
        </fieldset>
    );
}

MapForm.defaultProps = {
    coordinates: {
        lat: 0,
        lon: 0,
    },
    setCoordinates: () => {},
};

MapForm.propTypes = {
    coordinates: PropTypes.shape({
        lat: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
        lon: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    }).isRequired,
    setCoordinates: PropTypes.func.isRequired,
};

export default MapForm;
