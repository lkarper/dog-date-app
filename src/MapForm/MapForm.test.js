import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import MapForm from './MapForm';

describe('MapForm component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<MapForm />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<MapForm />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
