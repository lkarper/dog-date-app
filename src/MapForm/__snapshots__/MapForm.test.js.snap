// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapForm component renders the UI as expected 1`] = `
<MapForm
  coordinates={
    Object {
      "lat": 0,
      "lon": 0,
    }
  }
  setCoordinates={[Function]}
>
  <fieldset
    className="MapForm__coordinates-fieldset sub-fieldset"
  >
    <legend>
      Set coordinates for your meeting here
    </legend>
    <p>
      Using this feature will make it easier for others to find the location you set, but it is not required.
    </p>
    <Map
      setTempCoordinates={[Function]}
      tempCoordinates={
        Object {
          "lat": 0,
          "lon": 0,
        }
      }
    >
      <div
        className="Map__search-container"
      >
        <div
          className="Map__map-outer-container"
        >
          <ForwardGeocodeAddress
            setMarkerCoordinates={[Function]}
          >
            <div
              aria-live="polite"
              className="ForwardGeocodeAddress__outer-container"
            >
              <input
                aria-describedby="current-set-coordinates"
                aria-label="Search for a place by name or address."
                className="ForwardGeocodeAddress__search-input"
                id="place-search"
                name="place-search"
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Search by name or address"
                type="text"
                value=""
              />
              <NativeClickListener
                className="ForwardGeocodeAddress__click-container hidden"
                id="ForwardGeocodeAddress__click-container"
                onClick={[Function]}
              >
                <div
                  className="ForwardGeocodeAddress__click-container hidden"
                  id="ForwardGeocodeAddress__click-container"
                  role="alert"
                >
                  <ol
                    className="ForwardGeocodeAddress__results hidden"
                  />
                </div>
              </NativeClickListener>
              <div
                role="alert"
              />
            </div>
          </ForwardGeocodeAddress>
          <div
            className="Map__map-container"
            id="map-container"
          />
          <div
            className="Map__marker"
          />
          <div
            className="Map__coordinates"
            id="coordinates"
          >
            Latitude: 
            0
            , Longitude: 
            0
          </div>
        </div>
        <button
          className="Map__set-marker-button button"
          onClick={[Function]}
          type="button"
        >
          Set marker to my location
        </button>
        <p
          className="Map__marker-draggable-p"
        >
          (Marker is draggable.)
        </p>
      </div>
    </Map>
    <button
      className="MapForm__set-coordinates-button button"
      onClick={[Function]}
      type="button"
    >
      Set meeting place to above coordinates
    </button>
    <CurrentLocation
      coordinates={
        Object {
          "lat": 0,
          "lon": 0,
        }
      }
    >
      <div
        className="CurrentLocation__alert-div"
        role="alert"
      >
        <p
          id="current-set-coordinates"
        >
          No coordinates set (coordinates are optional).
        </p>
      </div>
    </CurrentLocation>
  </fieldset>
</MapForm>
`;
